### DISCUSSION TITLE
Github App Installation Setup URL does not preserve state query parameter

### DISCUSSION THREAD

**Discussion by sid374** [community/community#61291] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

We're creating a new Github App with the following expected installation flow:
1. We link to the github app installation like so: https://github.com/apps/sid-quickstart/installations/new?state=foo **(Notice the state=foo query parameter)**
2. The github app is configured with a setup url to our server
3. Once we get the redirect from github after the app install we look for the query parameter `state` in order to link the user identity to our system and protect against CSRF attacks

This is pretty standard oAuth stuff. However, we noticed that when Github redirects to our setup url, it does not preserver the `state` query parameter, and we're unable to complete the loop. I tried very hard but couldn't find any documentation about this either.

Interestingly if we link the user to the following installation url: https://github.com/apps/sid-quickstart/installations/select_target?state=foo - the `state` query param is correctly preserved when our setup url callback is called.

Can someone please help or clarify the expected behavior here?

**Comment by sid374** [comment]
@jamesmartin I've noticed that you've answered similar questions in the past - would you be able to help please?

**Comment by tihal** [comment]
+1 on this - seeing this behavior pop up just today. Our app installation and oauth flow was working yesterday, but today things are breaking because state is not being returned to our callback. Did something change on Github's end?

**Comment by handotdev** [comment]
+1 as well. Our GitHub apps are not working in production, and is affecting a lot of customers. Would really appreciate a fix asap

**Comment by coffeewithone** [comment]
+1 GitHub apps are not working and this is negatively affecting our customers! :(

**Comment by ojaswa1942** [comment]
+1
This affects our production as well.

**Comment by tamsalem** [comment]
+1 affects our prod, any ETA?

**Comment by livnoni** [comment]
Hi Github team, 
iI effect our production environment - Is there a time estimate for fixing the bug?

**Comment by hlevian** [comment]
+1
We are getting this error in our Production env and major customers, we'll appreciate a response here.

**Comment by kaje94** [comment]
+1
This is affecting us as well

**Comment by naposysdig** [comment]
+1

**Comment by richardguerre** [comment]
We have migrated to the new `/select_target` path and we would prefer not getting the rug pulled under us again (please don't change it back to `/new`).

**Reply by livnoni** [reply]
we also made the same fix.
please don't change to `/new` again.

**Comment by petebacondarwin** [comment]
Looks like Github have implemented a 302 redirect from /new to /select_target but failed to pass through the search params.

**Comment by FaizanNoor** [comment]
+1 on Production.

**Comment by alerosa** [comment]
+1

**Comment by kozliuk** [comment]
+1

**Comment by lyzohub** [comment]
+1

**Comment by hpsin** [comment, marked as answer]
Hey folks, thanks for reporting this. We have a fix in testing right now, about to be deployed. The `select_account` route is a new endpoint that's internal to the OAuth flow - please don't link to it. `/new` remains the correct endpoint.

**Reply by livnoni** [reply]
can u please update us exactly when the fix will be deployed ?

**Reply by mikeurbanski1** [reply]
+1

**Reply by HubbardJacob** [reply]
Thanks so much for the update!!!

**Reply by hpsin** [reply]
It's merging now, and I'll ping here when it's fully deployed.

**Reply by hpsin** [reply]
Alright, this is fixed, apologies for the impact here.

**Reply by RoxKilly** [reply]
This is possibly related to https://github.com/orgs/community/discussions/64239

**Comment by phantomcloud-io** [comment]
Any updates? It seems if the state is took long it wont pass

**Comment by mathvdc** [comment]
Seems the problem is back

**Reply by mathvdc** [reply]
https://github.com/orgs/community/discussions/125561

**Comment by vedant-contentstack** [comment]
Do we have any updates on the following???
